### PR TITLE
migrated-7.5/funding-options-wrong-sequence

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0.7  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

File: dspace_import.py [hunk 1]
```diff
@@ -1212,6 +1228,25 @@ def at_the_end_of_import():
...
+    separator = ';'
+
+    # sponsor list could have length 4 or 5
+    sponsor_list = wrong_sequence.split(separator)
+    org = sponsor_list[0]
+    project_code = sponsor_list[1]
+    project_name = sponsor_list[2]
+    project_type = sponsor_list[3]
+    eu_identifier = ''
+    if len(sponsor_list) == 5:
```
Comment (Paurikova2) [inline dspace_import.py:1243]: maybe use local var for const 5

**Review by Paurikova2 — CHANGES_REQUESTED** [review]

File: dspace_import.py [hunk 2]
```diff
@@ -97,13 +97,29 @@ def read_metadata():
...
+    metadatafield_json = read_json('metadatafieldregistry.json')
+    sponsor_field_id = -1
+    # Find out which field is `local.sponsor`, check only `sponsor` string
+    for i in metadatafield_json:
+        element = i['element']
+        if element != 'sponsor':
+            continue
+        sponsor_field_id = i['metadata_field_id']
+
+    print('sponsor_field_id: ' + str(sponsor_field_id))
```
[the remark this answers is not in the document]
Comment (milanmajchrak) [reply dspace_import.py:109]: Removed I don't want to log anything

File: dspace_import.py [hunk 3]
```diff
@@ -97,13 +97,29 @@ def read_metadata():
     """
     global metadatavalue
     metadatavalue_json = read_json('metadatavalue.json')
+    metadatafield_json = read_json('metadatafieldregistry.json')
+    sponsor_field_id = -1
+    # Find out which field is `local.sponsor`, check only `sponsor` string
+    for i in metadatafield_json:
```
[the remark this answers is not in the document]
Comment (milanmajchrak) [reply dspace_import.py:103]: Fixed

File: dspace_import.py [hunk 4]
```diff
@@ -1212,6 +1228,25 @@ def at_the_end_of_import():
     for key, value in statistics.items():
         logging.info(key + ": " + str(value[0]) + " expected and imported " + str(value[1]))
 
+# replace `local.sponsor` data sequence from `<ORG>;<PROJECT_CODE>;<PROJECT_NAME>;<TYPE>;<EU_IDENTIFIER>`
+# to `<TYPE>;<PROJECT_CODE>;<ORG>;<PROJECT_NAME>;<EU_IDENTIFIER>`
+def fix_local_sponsor_sequence(wrong_sequence):
```
[the remark this answers is not in the document]
Comment (milanmajchrak) [reply dspace_import.py:1233]: I've already added the comment, isn't that clear?

File: dspace_import.py (hunk 1 above)
Comment (milanmajchrak) [reply dspace_import.py:1243]: Fixed

File: dspace_import.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (milanmajchrak) [reply dspace_import.py:1237]: Ok, thank you

**Review by Paurikova2 — APPROVED** [review]

**Review by mrovnanik — APPROVED** [review]

**Pull request merged**